perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -271,81 +262,238 @@ def PercentileCalculator(numbers):
   return result
 
 
-def _ListObjects(storage_schema, bucket, prefix, host_to_connect=None):
-  """List objects under a bucket given a prefix.
+# ### Backends ###
 
-  Args:
-    storage_schema: The address schema identifying a storage. e.g., "gs"
-    bucket: Name of the bucket.
-    prefix: A prefix to list from.
-    host_to_connect: An optional endpoint string to connect to.
+class ObjectStorageBackend(object):
```

Comment:
Should we move this class and the backend implementation classes to their own files and probably under the one directory?

If we do that, we do need to change the high level benchmarking script in PKB to properly provision the test script on a VM (right now it just copies one file over).